source/kraken/apply_disruption.cpp

```diff
@@ -526,11 +526,14 @@ static auto comp = [](const impact_wptr& lhs, const impact_wptr& rhs){
     auto i_lhs = lhs.lock();
     auto i_rhs = rhs.lock();
 
+    // lexical sort by pointer then update datetime then uri
     if(!i_lhs || !i_rhs) {
-       return false;
+       return i_lhs.get() < i_rhs.get();
```

Comment:
better to do a set of shared_ptr, to avoid this test and be sure the comparator is stable.